src/lib.rs

```diff
@@ -83,6 +83,30 @@ impl<'a> Memchr<'a> {
             rposition: 0,
         }
     }
+
+    /// Get a reference to the haystack currently being iterated over.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # fn main() {
+    /// # use memchr::Memchr;
+    ///
+    /// let haystack = "Hello, world!\n, How are you\r\n";
+    /// let mut memchr = Memchr::new(b'\n', &haystack.as_bytes());
+    /// while let Some(pos) = memchr.next() {
+    ///     if let Some(&b'\r') = memchr.haystack().get(pos - 1) {
```

Comment:
This example doesn't do what it appears to do at first glance -- because `.haystack()` is changing during iteration.